extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/OidcBuildStep.java

```diff
@@ -243,8 +295,17 @@ public void registerTenantResolverInterceptor(Capabilities capabilities, OidcRec
             BuildProducer<SystemPropertyBuildItem> systemPropertyProducer) {
         if (!buildTimeConfig.auth.proactive
                 && (capabilities.isPresent(Capability.RESTEASY_REACTIVE) || capabilities.isPresent(Capability.RESTEASY))) {
-            var annotationInstances = combinedIndexBuildItem.getIndex().getAnnotations(TENANT_NAME);
-            if (!annotationInstances.isEmpty()) {
+            boolean foundTenantResolver = combinedIndexBuildItem
+                    .getIndex()
+                    .getAnnotations(TENANT_NAME)
+                    .stream()
+                    .map(AnnotationInstance::target)
+                    // ignored field injection points and injection setters
+                    // as we don't want to count in the TenantIdentityProvider injection point
+                    .filter(t -> t.kind() == METHOD)
```

Comment:
fyi @sberyozkin I didn't realize it, but now I remembered that the `@Tenant` annotation can also be used on classes: https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-tenancy/src/main/java/io/quarkus/it/keycloak/TenantEchoResource.java#L16

Test wasn't failing due to method-level instances detected there.

I'll tweak this condition next week to take this into a consideration. There is no hurry.

(will close the comment thread as this is just FYI)